B 3 implicit execution context

AndriiStefaniv opened the pull request:
#3

AndriiStefaniv committed:
Use implicit ExecutionContext

AndriiStefaniv committed:
Refactor tests for using implicit ExecutionContext

AndriiStefaniv committed:
Use current thread execution context for creation EMPTY_OR

NK91 commented on src/main/scala/scalty/types/error.scala:
```diff
@@ -29,11 +26,11 @@ object ErrorTypeExtensions {
   final class ServiceErrorExtension[T](val error: AppError) {
 
-    def toErrorOr: Or[T] = XorT.left[Future, AppError, T](Future.successful(error))
+    def toErrorOr(implicit ec: ExecutionContext): Or[T] = XorT.left[Future, AppError, T](Future.successful(error))
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/error.scala:
```diff
@@ -29,11 +26,11 @@ object ErrorTypeExtensions {
+    def toErrorOr(implicit ec: ExecutionContext): Or[T] = XorT.left[Future, AppError, T](Future.successful(error))
 
-    def toErrorOrWithType[D]: Or[D] = XorT.left[Future, AppError, D](Future.successful(error))
+    def toErrorOrWithType[D](implicit ec: ExecutionContext): Or[D] = XorT.left[Future, AppError, D](Future.successful(error))
```
Here are the same

NK91 commented on src/main/scala/scalty/types/error.scala:
```diff
@@ -29,11 +26,11 @@ object ErrorTypeExtensions {
+    def toErrorOrWithType[D](implicit ec: ExecutionContext): Or[D] = XorT.left[Future, AppError, D](Future.successful(error))
 
-    def toErrorOptionF: OptionF[T] = OptionT.none[Future, T]
+    def toErrorOptionF(implicit ec: ExecutionContext): OptionF[T] = OptionT.none[Future, T]
```
And here

NK91 commented on src/main/scala/scalty/types/optionF.scala:
```diff
@@ -24,11 +23,11 @@ trait OptionFTypeExtensions extends ScaltyExecutionContext {
 
   final class OptionFExtension[T](val value: T) {
-    def toOptionF: OptionF[T] = OptionT.pure[Future, T](value)
+    def toOptionF(implicit ec: ExecutionContext): OptionF[T] = OptionT.pure[Future, T](value)
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/optionF.scala:
```diff
@@ -24,11 +23,11 @@ trait OptionFTypeExtensions extends ScaltyExecutionContext {
 
   final class OptionTExtension[T](val value: Option[T]) {
-    def toOptionF: OptionF[T] = OptionT.fromOption[Future](value)
+    def toOptionF(implicit ec: ExecutionContext): OptionF[T] = OptionT.fromOption[Future](value)
```
Here are the same

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
 
   final class OrExtensions[T](val value: T) {
-    def toOr: Or[T] = XorT.right[Future, AppError, T](Future.successful(value))
+    def toOr(implicit ec: ExecutionContext): Or[T] = XorT.right[Future, AppError, T](Future.successful(value))
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
+    def toOr(implicit ec: ExecutionContext): Or[T] = XorT.right[Future, AppError, T](Future.successful(value))
 
-    def toEmptyOr: EmptyOr =
+    def toEmptyOr(implicit ec: ExecutionContext): EmptyOr =
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
 
   final class ListOrExtension[T](val value: List[T]) {
-    def toOr: Or[List[T]] =
+    def toOr(implicit ec: ExecutionContext): Or[List[T]] =
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
 
   final class FoldableExtension[T](val values: List[Or[T]]) {
-    def foldable: Or[List[T]] =
+    def foldable(implicit ec: ExecutionContext): Or[List[T]] =
```
maybe replace Foldable to "traverse" ?

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
 
   final class OptionOrExtension[T](val option: Option[T]) {
-    def toOptionOr: OptionOr[T] = OptionT.fromOption[Or](option)
+    def toOptionOr(implicit ec: ExecutionContext): OptionOr[T] = OptionT.fromOption[Or](option)
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
+    def toOptionOr(implicit ec: ExecutionContext): OptionOr[T] = OptionT.fromOption[Or](option)
 
-    def toNoneOr: Or[Option[T]] = XorT.right[Future, AppError, Option[T]](Future.successful(None))
+    def toNoneOr(implicit ec: ExecutionContext): Or[Option[T]] = XorT.right[Future, AppError, Option[T]](Future.successful(None))
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
+    def toNoneOr(implicit ec: ExecutionContext): Or[Option[T]] = XorT.right[Future, AppError, Option[T]](Future.successful(None))
 
-    def toNoneOrWith[D]: Or[Option[D]] = XorT.right[Future, AppError, Option[D]](Future.successful(None))
+    def toNoneOrWith[D](implicit ec: ExecutionContext): Or[Option[D]] = XorT.right[Future, AppError, Option[D]](Future.successful(None))
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -84,70 +86,70 @@ object OrTypeExtensions {
+    def toNoneOrWith[D](implicit ec: ExecutionContext): Or[Option[D]] = XorT.right[Future, AppError, Option[D]](Future.successful(None))
 
-    def toOrWithLeftError(error: AppError): Or[T] = option match {
+    def toOrWithLeftError(error: AppError)(implicit ec: ExecutionContext): Or[T] = option match {
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -156,43 +158,43 @@ object OrTypeExtensions {
 
   final class OptionOrValueExtension[T](val value: T) {
-    def toOptionOr: OptionOr[T] = OptionT.fromOption[Or](Option(value))
+    def toOptionOr(implicit ec: ExecutionContext): OptionOr[T] = OptionT.fromOption[Or](Option(value))
```
It doesn't need implicit executionContext, better to use currentThreadExecutionContext

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -204,9 +206,14 @@ object OrTypeExtensions {
 
-  implicit def xorTFMonoid[T]: Monoid[Or[List[T]]] = new Monoid[Or[List[T]]] {
+  val EMPTY_OR: EmptyOr = {
+    import scalty.context.ScaltyExecutionContext.currentThreadExecutionContext
```
better change to direct pass argument than implicit params

AndriiStefaniv committed:
Apply conversations comments of pull request #4

Pull request https://github.com/awesome-it-ternopil/scalty/pull/4.
Comments:
- remove using ExecutionContext with creation new thread for operation with `Future.successful`
- impl method `foldable` via traverse

AndriiStefaniv committed:
Merge branch 'master' into b_3_implicit_execution_context

# Conflicts:
#	src/main/scala/scalty/types/error.scala

NK91 commented on src/main/scala/scalty/types/or.scala:
```diff
@@ -106,27 +105,28 @@ object OrTypeExtensions {
   final class FoldableExtension[T](val values: List[Or[T]]) {
     def foldable(implicit ec: ExecutionContext): Or[List[T]] =
-      Foldable[List].foldMap(values)(a => a.map(List(_)))(or.xorTFMonoid[T])
+      values.traverse[Or, T](or => or)
```
or => or replace to identity ?
